tools/c7n_azure/c7n_azure/query.py

```diff
@@ -28,30 +29,11 @@ def __init__(self, session_factory):
 
     def filter(self, resource_manager, **params):
         m = resource_manager.resource_type
-        client = local_session(self.session_factory).client(
-            "%s.%s" % (m.service, m.client))
         enum_op, list_op = m.enum_spec
-        op = getattr(getattr(client, enum_op), list_op)
-        data = [self.to_dictionary(e) for e in op()]
-        return data
+        op = getattr(getattr(resource_manager.get_client(), enum_op), list_op)
+        data = [r.serialize(True) for r in op()]
```

Comment:
not for this pr, but worth looking to see if we can avoid the deserialize into objects in the first place via appropriate config of the the msrest layer, that would also avoid the key mangle.

Reply:
#2231 agreed.  Looked into this a bit a few weeks ago and I believe possible but it is non-obvious exactly how to pull it off, we'll just bring it up next time we chat with the SDK dev team.